src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -16,6 +17,14 @@
           Condition="'$(RestorePackages)'=='true' AND '$(IsTestProject)' == 'true' AND '$(TestRuntimeProjectJson)' != ''">
     <Exec Command="$(DnuRestoreCommand) &quot;$(TestRuntimeProjectJson)&quot;" StandardOutputImportance="Low" CustomErrorRegularExpression="^Unable to locate .*" />
   </Target>
+  
+  <Target Name="ValidateTestRuntimeDependencies"
+          AfterTargets="RestoreTestRuntimePackage"
```

Comment:
The way I've got it going now without semaphores (working in my Razzle build and a git corefx repo) is that a BuildAllProjects build will run a `ValidateAllProjectDependencies` target (similar to `BatchRestorePackages`), and the code in buildtools is skipped. The validation targets in buildtools only run when `RestorePackages=true`, which in corefx's case is during an individual project build.

I don't know if leaving it all to a `BatchRestorePackages`-like step is great, since someone working on and building only one project would miss test-runtime and tool-runtime validation until they (or CI) do an all-project build.

On the other hand, we could validate all project.jsons in the entire repository on every build. Validation is pretty fast, so maybe that simplicity vs. speed tradeoff is good. What do you think?

(The `TestRuntimeSemaphore` in this PR was just a fix to avoid restoring more than necessary, and doesn't affect the validation logic. I'll split off that change to keep things pure.)